addons/t9n/models/resource.py

```diff
@@ -14,3 +18,38 @@ class Resource(models.Model):
     project_id = fields.Many2one(
         comodel_name="t9n.project",
     )
+
+    def _decode_resource_file(self, resource_file):
+        try:
+            file_content = base64.b64decode(resource_file).decode("utf-8")
+            po_obj = polib.pofile(file_content)
+        except (IOError, UnicodeDecodeError):
+            raise ValidationError(
+                _("The file should be a .po file with a valid syntax."))
+        return po_obj
+
+    def _define_create_commands(self, resource_file):
+        messages_to_create = [{"body": entry.msgid, "context": entry.msgctxt,
+                               "translator_comments": entry.tcomment, "extracted_comments": entry.comment, "reference_ids": [Command.create({"body": reference[0] + reference[1]}) for reference in entry.occurrences]} for entry in resource_file]
+        return [Command.create(message_data) for message_data in messages_to_create]
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if not vals["file"]:
```

Comment:
```suggestion
            if not vals.get("file"):
```
To avoid an exception if file is not in vals